x-pack/plugins/index_management/server/routes/api/inference_models/register_get_route.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { InferenceAPIConfigResponse } from '@kbn/ml-plugin/common/types/trained_models';
```

Comment:
Once you move this type to the ml trained models package, import it from there here as well.